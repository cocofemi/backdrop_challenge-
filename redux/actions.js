import * as types from './actionTypes';

export const add_favorite = (data) => dispatch => {
    dispatch({
        type: types.ADD_FAVORITE,
        payload: data
    })
}   

export const remove_favorite = (id) => dispatch => {
    dispatch({
        type: types.REMOVE_FAVORITE,
        payload:{
            id
        }
    })
}