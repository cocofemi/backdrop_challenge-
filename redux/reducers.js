import * as types from './actionTypes'

const initialState = {
    favorites: []
}
export default function favoriteReducer (state = initialState, action) {
    switch(action.type) {
        case types.ADD_FAVORITE: 
            return{...state, favorites:[...state.favorites, action.payload]}
        case types.REMOVE_FAVORITE:
            return{
                ...state,
                favorites: state.favorites.filter(favorite => favorite.id !== action.payload.id)
            }
        default: 
            return state
    }
}