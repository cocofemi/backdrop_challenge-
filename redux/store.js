import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import favoriteReducer from './reducers';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { persistStore, persistReducer } from 'redux-persist';
import {STORAGE_KEY} from "@env";

const persistConfig = {
    key: STORAGE_KEY,
    storage: AsyncStorage,
    whitelist: ['favorites']
}

const rootReducer = combineReducers({
    favorite: persistReducer(persistConfig, favoriteReducer)
})

export const store = createStore(rootReducer, applyMiddleware(thunk));
export const persistor = persistStore(store);