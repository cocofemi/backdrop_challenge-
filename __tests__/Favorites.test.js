import React  from 'react';
import { cleanup, render, waitFor } from '@testing-library/react-native';
import fetchMock from "jest-fetch-mock"
import { Provider } from 'react-redux';
import { store } from '../redux/store';
import Favorites from '../screens/Favourites'

fetchMock.enableMocks();
afterEach(cleanup);

describe('Testing FavoritesPage', () => {
    it('should render the favorite page correctly', () => {
      const tree = render(<Provider store={store}><Favorites /></Provider>)
      expect(tree).toMatchSnapshot();
    }),
    it('should render a list of favorite cats', async() => {
      fetch.mockResponseOnce(
        JSON.stringify([
          { name: 'bengal', image: 'https://cdn2.thecatapi.com/images/ozEvzdVM-.jpg', id: 'beng' },
          { name: 'Aegean', image: 'https://cdn2.thecatapi.com/images/ozEvzdVM-.jpg', id: 'aege' },
        ])
      )
  
      const {getByTestId } = render(<Provider store={store}><Favorites /></Provider>)
      expect(getByTestId('favorite-row-0'));
    })   
  })