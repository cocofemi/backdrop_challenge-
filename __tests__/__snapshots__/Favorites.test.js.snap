// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FavoritesPage should render the favorite page correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 1,
      "width": 400,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "600",
        "marginLeft": 25,
        "marginTop": 80,
      }
    }
  >
    Cats I Like
  </Text>
  <RCTScrollView
    data={Array []}
    extraData={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="favorite-row-0"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
